test/clients/openshift/login.go

```diff
@@ -46,69 +46,76 @@ func login(username string) (*api.Config, error) {
 		return nil, fmt.Errorf("unknown username %q", username)
 	}
 
-	pool := x509.NewCertPool()
-	pool.AddCert(cs.Config.Certificates.Ca.Cert)
-
-	cli := &http.Client{
-		Transport: &http.Transport{
-			TLSClientConfig: &tls.Config{
-				RootCAs: pool,
-			},
+	params := azuretls.CertParams{
+		Subject: pkix.Name{
+			CommonName:   username,
+			Organization: organization,
 		},
-		CheckRedirect: func(req *http.Request, via []*http.Request) error {
-			return http.ErrUseLastResponse
-		},
-	}
-
-	req, err := http.NewRequest(http.MethodGet, "https://"+cs.Properties.FQDN+"/oauth/authorize?response_type=token&client_id=openshift-challenging-client", nil)
-	req.Header.Add("X-Csrf-Token", "1")
-	req.SetBasicAuth(username, password)
-
-	resp, err := cli.Do(req)
-	if err != nil {
-		return nil, err
+		ExtKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
+		SigningCert: cs.Config.Certificates.Ca.Cert,
+		SigningKey:  cs.Config.Certificates.Ca.Key,
 	}
-
-	location, err := resp.Location()
+	key, cert, err := azuretls.NewCert(&params)
 	if err != nil {
 		return nil, err
 	}
 
-	fragment, err := url.ParseQuery(location.Fragment)
+	kc, err := makeKubeConfig(key, cert, cs.Config.Certificates.Ca.Cert, cs.Properties.FQDN, username, "default")
+	var c api.Config
+	err = latest.Scheme.Convert(kc, &c, nil)
 	if err != nil {
 		return nil, err
 	}
-
-	return makeKubeConfig(cs.Config.Certificates.Ca.Cert, cs.Properties.FQDN, username, fragment.Get("access_token"))
+	return &c, nil
 }
 
-func makeKubeConfig(caCert *x509.Certificate, endpoint, username, token string) (*api.Config, error) {
+func makeKubeConfig(clientKey *rsa.PrivateKey, clientCert, caCert *x509.Certificate, endpoint, username, namespace string) (*v1.Config, error) {
```

Comment:
changing the signature to return a *v1.Config instead of an *api.Config introduces another conversion - why is it necessary to do this?

Reply:
done

Reply:
huh? can't this go back to returning an *api.Config?